NWN.Anvil/src/main/API/Objects/NwCreature.cs

```diff
@@ -993,6 +1009,33 @@ public async Task ActionCounterspell(NwGameObject counterSpellTarget)
       NWScript.ActionCounterSpell(counterSpellTarget);
     }
 
+    /// <summary>
+    /// Forces a PC to level up without the level up GUI.
+    /// </summary>
+    /// <param name="nClass">The class to level up.</param>
+    /// <param name="nHitDie">The hit points gained during this level up.</param>
+    /// <param name="nAbilityGain">The abilty to increase : 6 = no increase.</param>
+    /// <param name="bEpic">Is this an epic level.</param>
+    /// <param name="nSkillPointsRemaining">Number of skill points to give</param>
+    /// <param name="nDomain1">Sets a cleric domain for the class (255 = no domain)</param>
+    /// <param name="nDomain2">Sets the second cleric domain for the class (255 = no domain)</param>
+    /// <param name="nSchool">Sets the wizard school for the class (255 = no school)</param>
+    /// <param name="addStatsToList">Adds the new stats to the character sheet</param>
+    public void ForceLevelUp(byte nClass, byte nHitDie, byte nAbilityGain = 6, int bEpic = 0, ushort nSkillPointsRemaining = 0, byte nDomain1 = 255, byte nDomain2 = 255, byte nSchool = 255, int addStatsToList = 1)
+    {
+      CNWLevelStats stats = new CNWLevelStats()
+      {
+        m_nClass = nClass,
+        m_nHitDie = nHitDie,
+        m_nAbilityGain = nAbilityGain,
+        m_bEpic = bEpic,
+        m_nSkillPointsRemaining = nSkillPointsRemaining,
+      };
+
+      Creature.m_pStats.LevelUp(stats, nDomain1, nDomain2, nSchool, addStatsToList);
+      GC.SuppressFinalize(stats);
+    }
```

Comment:
Anvil API functions should use wrapper classes and more friendly types when possible.

We do not prefix parameter names with the type.

```suggestion
    public void ForceLevelUp(NwClass classType, byte hitDie, Ability? abilityGain = default, bool epic = false, ushort skillPointsRemaining = 0, NwDomain? domain1 = default, NwDomain? domain2 = default, SpellSchool school = SpellSchool.Unknown, bool addStatsToList = true)
    {
      CNWLevelStats stats = new CNWLevelStats()
      {
        m_nClass = classType.Id,
        m_nHitDie = hitDie,
        m_nAbilityGain = abilityGain?.Value ?? 6,
        m_bEpic = epic.ToInt(),
        m_nSkillPointsRemaining = skillPointsRemaining,
      };

      Creature.m_pStats.LevelUp(stats, nDomain1?.Id ?? 255, nDomain2?.Id ?? 255, (unchecked((byte)School), addStatsToList.ToInt());
      GC.SuppressFinalize(stats);
    }
```

Please update the comments with the parameter changes :)